node/repo/memrepo.go

```diff
@@ -107,6 +107,12 @@ func (lmem *lockedMemRepo) Path() string {
 	}
 
 	if lmem.t == StorageMiner {
+		// this is required due to the method makeDealStaging from cmd/lotus-storage-miner/init.go
+		// deal-staging is the directory deal files are staged in before being sealed into sectors
+		// for offline deal flow.
+		if err := os.MkdirAll(filepath.Join(t, "deal-staging"), 0755); err != nil {
```

Comment:
This is kinda ugly, but I didn't want to make the Repo responsible for creating this directory since it is used by things other than the miner.